test/snapshots/conftest.py

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/python3
+
+from test.pylib.cql_repl import conftest
```

Comment:
There is absolutely no need to create yet another top-level test directory. See https://github.com/scylladb/scylladb/issues/16026

The test you wrote looks like it can fit right into the existing test/topology_custom. 

Even better would have been to put it in test/nodetool/test_snapshot.py, which is a much more elegant test framework and doesn't force you to create a Scylla server just to test one request.

Reply:
Please take a look at test/cql-pytest/README.md for an overview of how the cql-pytest (and its cousin test/rest_api) work, and I'll be happy to explain it further if you have specific questions (we can also do on one of the mailing lists, here seems pretty odd.

To make a long story short, the tests in test/rest_api (and also in test/cql-pytest and test/alternator) all connect to one single-node cluster - this single node is started once, and then potentially a thousand different tests are run on it. This is how test/alternator, for example, runs around a thousand tests in just 30 seconds - on a single code :-) How does this server start? Well, you can start one yourself and then run the test with "pytest", but the test.py script and also the test/rest_api/run script automate the starting of the Scylla node and then running the test against it.

Normally, the test only connects to Scylla using TCP - to the CQL port, DynamoDB port, or REST API port (this is what you need to get a snapshot). But how does the test know where these snapshots are stored? There are two answers to that:

1. It's possible. In test/cql-pytest we have some cool (I think) example in test/cql-pytest/test_logs.py of logfile_path() of finding where the log file of the Scylla to which you connected lives.
2. BUT, you probably don't need to do it! The existing snapshot tests in test/rest_api/test_storage_service.py don't need to find where on disk these snapshots live - they first check if the snapshot REST API request succeeds or fails. Then they also have another REST API to _list_ the snapshots successfully created. They don't need to bother to find these snapshots on disk. In theory - and in practice (!) - we can run this and most other tests against Scylla Cloud - some remote Scylla installation that we can reach via CQL and REST API but can't actually read its files.

If this doesn't fully answer your question, please don't hesitate to ask - I would really like to help developers write more tests and better tests, and these simple and fast frameworks (test/cql-pytest, test/rest_api, test/alternator) are, in my opinion, an important thing to make tests easier to write and encourage everyone to write more tests.

Finally, don't forget that if you don't like test/rest_api and like to keep the test exactly as you wrote it, it's also possible - just put it (I believe) in test/topology_custom and not in a new directory.

Reply:
Lemme check how it feets

Reply:
Glad you mentioned test_storage_service, apparently it already have all needed tests, thanks to @bhalevy :)